servers/navigation/navigation_path_query_parameters_3d.h

```diff
@@ -60,22 +59,32 @@ class NavigationPathQueryParameters3D : public RefCounted {
 		PATH_METADATA_INCLUDE_ALL = NavigationUtilities::PathMetadataFlags::PATH_INCLUDE_ALL
 	};
 
-	const NavigationUtilities::PathQueryParameters &get_parameters() const { return parameters; }
+private:
+	PathfindingAlgorithm pathfinding_algorithm = PATHFINDING_ALGORITHM_ASTAR;
+	PathPostProcessing path_postprocessing = PATH_POSTPROCESSING_CORRIDORFUNNEL;
+	RID map;
+	Vector3 start_position;
+	Vector3 target_position;
+	uint32_t navigation_layers = 1;
+	BitField<PathMetadataFlags> metadata_flags = PATH_METADATA_INCLUDE_ALL;
+	bool simplify_path = false;
+	real_t simplify_epsilon = 0.0;
```

Comment:
These internals seems really similar to the 2d ones. Maybe have it in a struct (except start and target if it has to be 2D for the 2D version, but he could probably store it 3D, and forward them only 2D)

Reply:
They are similar but I am not sure about moving them to a shared struct. We don't really do this for other shared class properties as it adds code complexity without a strong need for it.